src/main/java/io/gravitee/policy/groovy/GroovyPolicy.java

```diff
@@ -56,6 +57,14 @@ public class GroovyPolicy extends GroovyPolicyV3 implements Policy {
     public GroovyPolicy(GroovyPolicyConfiguration configuration) {
         super(configuration);
         scriptFlowable = Flowable.fromIterable(configuration.getScripts());
+
+        // Precompile all the scripts on a io schedulers to get ready when necessary.
+        scriptFlowable
+            .doOnNext(GROOVY_SHELL::compile)
+            .subscribeOn(Schedulers.io())
+            .doOnError(e -> log.warn("Error while compiling script. Ignoring", e))
+            .onErrorComplete()
+            .subscribe();
```

Comment:
I've checked and it uses a cache that awaits the result of a pending load (so only one compilation). This will be on the schedulers.io() so no risk of blocking the event loop.